public/app/features/alerting/unified/components/rule-editor/AnnotationsStep.tsx

```diff
@@ -145,7 +145,10 @@ const AnnotationsStep = () => {
                       <ValueInputComponent
                         data-testid={`annotation-value-${index}`}
                         className={cx(styles.annotationValueInput, { [styles.textarea]: !isUrl })}
-                        {...register(`annotations.${index}.value`)}
+                        {...register(`annotations.${index}.value`, {
+                          validate: (value: string) =>
+                            isUrl && !value.startsWith('http') ? 'URL must start with http' : true,
```

Comment:
I don't think it's "starts with http", as it could be `file://`, `/foo`, `ftp://`...
We might want to do a combination of `try new URL`, and `starts with /`? Or similar

Reply:
I was not sure about this, but if we allow these kind of urls, maybe only using new URL will be enough to cover all the cases

Reply:
I would turn this into a separate function that checks for either a relative or absolute URL (using `try new URL`).

Let's also keep in mind that these rules _may_ have been created via the API or provisioning where this kind of validation is not enforced.

With that in mind, should we prevent users from updating alert rules that may have been created via the API with broken URLs?